mathics/core/rules.py

```diff
@@ -94,6 +94,8 @@ def __init__(self, pattern, replace, system=False):
     def do_replace(self, vars, options, evaluation):
         new = self.replace.replace_vars(vars)
         new.options = options
+        if len(options) > 0:  # need to trigger (re)evaluation?
```

Comment:
Yes, that's correct, I added a comment in that code, trying to explain what happens there. It's related to the way `OptionValue[]`s are resolved: it happens through applying `OptionValue.apply` and that needs `evaluate()` to get called on the expression in the first place. Another way to fix this might be to change `if self.is_evaluated` in `Expression.evaluate()` to  `if self.is_evaluated and not self.options`, in order to always evaluate if options need to be applied to the expression, but I'm much less confident in the side effects this might have.